bootstrap/cmd/bootstrap/app/gcpUtils.go

```diff
@@ -279,3 +282,31 @@ func (s *ksServer) ApplyIamPolicy(ctx context.Context, req ApplyIamRequest) erro
 	}
 	return nil
 }
+
+func (s *ksServer) SyncPersistentResource(ctx context.Context, req SyncResourceRequest) error {
+	resoureRepo := fmt.Sprintf("%s-persistent-resource", req.Project)
+	err := s.CloneRepoToLocal(req.Project, req.Token, resoureRepo, true)
+	if err != nil {
+		return err
+	}
+	secretDir := path.Join(s.appsDir, resoureRepo, req.Name, "secret")
+	_, err = s.fs.Stat(secretDir)
+	if err == nil {
+		os.RemoveAll(secretDir)
+	}
+	os.Mkdir(secretDir, os.ModePerm)
+	k8sConfig, err := buildClusterConfig(ctx, req.Token, req.Project, req.Zone, req.Cluster)
+	k8sClientset, err := clientset.NewForConfig(k8sConfig)
+	for _, sec := range []string{"envoy-ingress-tls", "letsencrypt-prod-secret"} {
+		secret, err := k8sClientset.CoreV1().Secrets(req.Namespace).Get(sec, meta_v1.GetOptions{})
+		if err != nil {
+			DumpConfig(path.Join(secretDir, sec), secret)
```

Comment:
Did you mean to do this in the error case?
Return or warn on the error instead?

Reply:
Good catch